tests/ui/traits/next-solver/coherence/skip-reporting-if-references-err.rs

```diff
@@ -0,0 +1,15 @@
+//@ compile-flags: -Znext-solver
+
```

Comment:
```suggestion
// Regression test for #121006.
```

Reply:
also: slightly minimized
```rust
trait ToUnit<'a> {
    type Unit;
}

impl<T> ToUnit for T {}
//~^ ERROR implicit elided lifetime not allowed here

trait Overlap {}
impl<U> Overlap for fn(U) {}
impl Overlap for for<'a> fn(<() as ToUnit<'a>>::Unit) {}

fn main() {}
```

Please move it to the general `coherence` folder, it also affects the old solver (and use revisions)